rust/main/submitter/src/chain_tx_adapter/chains/ethereum/payload.rs

```diff
@@ -0,0 +1,15 @@
+use ethers::abi::Function;
+use ethers::types::transaction::eip2718::TypedTransaction;
+
+use crate::payload::FullPayload;
+
+pub(crate) trait Precursor {
```

Comment:
this trait is a duplicate of the one in the sealevel adapter, we should move it out of the vm-specific adapters and reuse it